test/Network/IpAddress.cpp

```diff
@@ -0,0 +1,223 @@
+        }
+
+        SUBCASE("operator<")
+        {
+            CHECK(sf::IpAddress(1) < sf::IpAddress(2));
+            CHECK(sf::IpAddress() < sf::IpAddress(0, 0, 0, 0));
+            CHECK(sf::IpAddress(0, 0, 0, 0) < sf::IpAddress(1, 0, 0, 0));
+            CHECK(sf::IpAddress(1, 0, 0, 0) < sf::IpAddress(0, 1, 0, 0));
+            CHECK(sf::IpAddress(0, 1, 0, 0) < sf::IpAddress(0, 0, 1, 0));
+            CHECK(sf::IpAddress(0, 0, 1, 0) < sf::IpAddress(0, 0, 0, 1));
+            CHECK(sf::IpAddress(0, 0, 0, 1) < sf::IpAddress(1, 0, 0, 1));
+        }
+
+        SUBCASE("operator>")
+        {
+            CHECK(sf::IpAddress(2) > sf::IpAddress(1));
+            CHECK(sf::IpAddress(0, 0, 0, 0) > sf::IpAddress());
+            CHECK(sf::IpAddress(1, 0, 0, 0) > sf::IpAddress(0, 0, 0, 0));
+            CHECK(sf::IpAddress(0, 1, 0, 0) > sf::IpAddress(1, 0, 0, 0));
+            CHECK(sf::IpAddress(0, 0, 1, 0) > sf::IpAddress(0, 1, 0, 0));
+            CHECK(sf::IpAddress(0, 0, 0, 1) > sf::IpAddress(0, 0, 1, 0));
+            CHECK(sf::IpAddress(1, 0, 0, 1) > sf::IpAddress(0, 0, 0, 1));
+        }
+
+        SUBCASE("operator<=")
+        {
+            CHECK(sf::IpAddress(1) <= sf::IpAddress(2));
+            CHECK(sf::IpAddress() <= sf::IpAddress(0, 0, 0, 0));
+            CHECK(sf::IpAddress(0, 0, 0, 0) <= sf::IpAddress(1, 0, 0, 0));
+            CHECK(sf::IpAddress(1, 0, 0, 0) <= sf::IpAddress(0, 1, 0, 0));
+            CHECK(sf::IpAddress(0, 1, 0, 0) <= sf::IpAddress(0, 0, 1, 0));
+            CHECK(sf::IpAddress(0, 0, 1, 0) <= sf::IpAddress(0, 0, 0, 1));
+            CHECK(sf::IpAddress(0, 0, 0, 1) <= sf::IpAddress(1, 0, 0, 1));
+
+            CHECK(sf::IpAddress() <= sf::IpAddress());
+            CHECK(sf::IpAddress("") <= sf::IpAddress(""s));
+            CHECK(sf::IpAddress("8.8.8.8") <= sf::IpAddress(8, 8, 8, 8));
+            CHECK(sf::IpAddress(0x42, 0x69, 0x96, 0x24) <= sf::IpAddress(0x42699624));
+            CHECK(sf::IpAddress(0xABCDEF01) <= sf::IpAddress("171.205.239.1"));
+        }
+
+        SUBCASE("operator>=")
+        {
+            CHECK(sf::IpAddress(2) >= sf::IpAddress(1));
+            CHECK(sf::IpAddress(0, 0, 0, 0) >= sf::IpAddress());
+            CHECK(sf::IpAddress(1, 0, 0, 0) >= sf::IpAddress(0, 0, 0, 0));
+            CHECK(sf::IpAddress(0, 1, 0, 0) >= sf::IpAddress(1, 0, 0, 0));
+            CHECK(sf::IpAddress(0, 0, 1, 0) >= sf::IpAddress(0, 1, 0, 0));
+            CHECK(sf::IpAddress(0, 0, 0, 1) >= sf::IpAddress(0, 0, 1, 0));
+            CHECK(sf::IpAddress(1, 0, 0, 1) >= sf::IpAddress(0, 0, 0, 1));
+
+            CHECK(sf::IpAddress() >= sf::IpAddress());
+            CHECK(sf::IpAddress("") >= sf::IpAddress(""s));
+            CHECK(sf::IpAddress("8.8.8.8") >= sf::IpAddress(8, 8, 8, 8));
+            CHECK(sf::IpAddress(0x42, 0x69, 0x96, 0x24) >= sf::IpAddress(0x42699624));
+            CHECK(sf::IpAddress(0xABCDEF01) >= sf::IpAddress("171.205.239.1"));
+        }
```

Comment:
I have the feeling that we might want to test all relational operators in a generic way in the future. This would reduce the amount of test code to write and enforce internal consistency between the operators.

Something like this:
```cpp
#define CHECK_LESS(lower, upper) \
    CHECK(lower < upper);        \
    CHECK(upper > lower);        \
    CHECK(!(lower >= upper));    \
    CHECK(!(upper <= lower));
```

Reply:
It ought to be a macro to preserve line number information.

Reply:
Sounds great, thanks @ChrisThrasher!

I think macros have too bad of a reputation. Yes, they shouldn't be used for `MAX(a, b)` and other cases where existing abstraction mechanisms work just fine, but they have their niche where nothing comes close.